servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -5026,6 +5077,15 @@ public void putPolicy(ResourceContext ctx, String domainName, String policyName,
 
         validatePolicyAssertions(policy.getAssertions(), caller);
 
+        // we are not going to allow any user to add/update a policy
+        // that one of its assertions has role associated that does not exist
+
+        RoleList roleList = getRoleList(ctx, domainName, null, null);
```

Comment:
Same comment here as above. getRoleList is an API endpoint. Let's replace this with an internal implementation and not call another API endpoint.